src/qibo/quantum_info/random_ensembles.py

```diff
@@ -363,9 +384,11 @@ def random_density_matrix(
         pure (bool, optional): if ``True``, returns a pure state. Default is ``False``.
         metric (str, optional): metric to sample the density matrix from. Options:
             ``"Hilbert-Schmidt"`` and ``"Bures"``. Default is ``"Hilbert-Schmidt"``.
-        basis (str, optional): if ``"pauli"``, return random density matrix in the
-            Pauli basis. If ``None``, returns it in the computational basis.
-            Default is ``None``.
+        basis (str, optional): if ``None``, returns random density matrix in the
+            computational basis. If ``"pauli"``, returns it in the Pauli basis.
+            If "pauli-<pauli_order>", (e.g. "pauli-IZXY"), returns it in the Pauli basis
+            with the corresponding order of single-qubit Pauli elements
+            (see :func:`qibo.quantum_info.pauli_basis`). Default is ``None``.
```

Comment:
If the only possible basis are `None` (for the computational basis) and `pauli-<pauli_order>`, why don't you ask for the `pauli_order` directly, as you are doing in all other functions? And if it is `None` you can fall back to the computational.

This way you will avoid the complicated string manipulations and checks in the function's body. Unless you are planning to extend this to other basis or I am missing something.

On another note, not related to this PR, I see that type annotations are used in the function arguments. If you want to be strict about the types, if an argument can be `None`, you should use
```py
from typing import Optional

rank: Optional[int] = None
```
instead of 
```
rank: int = None
```
but as I said this is not required for this PR.

Reply:
I liked the first suggestion and implemented it.

Also agree with the second one, but will do in a different PR.